x/finality/types/msg.go

```diff
@@ -20,6 +20,30 @@ func (m *MsgAddFinalitySig) MsgToSign() []byte {
 	return msgToSignForVote(m.BlockHeight, m.BlockAppHash)
 }
 
+func (m *MsgAddFinalitySig) ValidateBasic() error {
```

Comment:
We could check lengths of `fpBtcPk`, `PubRand`, and `FinalitySig`

Reply:
good point! 👍